android/src/main/java/com/rnfs/RNFSManager.java

```diff
@@ -759,6 +765,80 @@ public void touch(String filepath, double mtime, double ctime, Promise promise)
     }
   }
 
+  @ReactMethod
+  public void canOpenFile(String path, String mime, final Promise promise) {
+    try {
+      Uri uriForFile = FileProvider.getUriForFile(getCurrentActivity(),
+              this.getReactApplicationContext().getPackageName() + ".provider", new File(path));
+
+      // Create the intent with data and type
+      Intent intent = new Intent(Intent.ACTION_VIEW)
+              .setDataAndType(uriForFile, mime);
+
+      // Set flag to give temporary permission to external app to use FileProvider
+      intent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+        // All the activity to be opened outside of an activity
+      intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+      // Validate that the device can open the file
+      PackageManager pm = getCurrentActivity().getPackageManager();
+      if (intent.resolveActivity(pm) != null) {
+        promise.resolve(true);
+      }
+      else {
+        promise.resolve(false);
+      }
+
+    } catch(Exception ex) {
+      promise.reject("EUNSPECIFIED", ex.getLocalizedMessage());
+    }
+  }
+
+  @ReactMethod
+  public void openFile(String path, String mime, final Promise promise) {
+    try {
+      Uri uriForFile = FileProvider.getUriForFile(getCurrentActivity(),
+              this.getReactApplicationContext().getPackageName() + ".provider", new File(path));
+
+      // Create the intent with data and type
+      Intent intent = new Intent(Intent.ACTION_VIEW)
+              .setDataAndType(uriForFile, mime);
+
+      // Set flag to give temporary permission to external app to use FileProvider
+      intent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+        // All the activity to be opened outside of an activity
+      intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+      // Validate that the device can open the file
+      PackageManager pm = getCurrentActivity().getPackageManager();
+      if (intent.resolveActivity(pm) != null) {
+        this.getReactApplicationContext().startActivity(intent);
+      }
+
+      ActionViewVisible = true;
```

Comment:
No. Not sure why this was done this way (which edge case it was supposed to solve). But the behaviour is ok: when `onHostResume` triggers we want to resolve the promise. 

No idea why this check is needed, but let's keep it in just in case it's important. Or do you think we should drop it? 
```
if(ActionViewVisible)
            promise.resolve(null);
```

Context doesn't explain much:  https://github.com/wkh237/react-native-fetch-blob/issues/165 https://github.com/joltup/rn-fetch-blob/commit/c6119a63c385e598794cb0f6882e047f19663a4d https://github.com/wkh237/react-native-fetch-blob/commit/8215f072a3f81848887f8028d9b0917314b4cdea

Reply:
We could just return `promise.resolve(null);` instead, since we don't need to know when users come back atm

Reply:
I dropped it & went with https://github.com/ExodusMovement/react-native-fs/pull/5/commits/6e690fede2fa88bc1b5fe35aabc728a3182b7b72